_stbt/core.py

```diff
@@ -2991,25 +2989,41 @@ def test_wait_for_motion_half_motion_int():
 
 @contextmanager
 def _fake_frames_at_half_motion():
-    class FakeDisplay(object):
-        def frames(self, _timeout_secs=10):
-            data = [
-                numpy.zeros((2, 2, 3), dtype=numpy.uint8),
-                numpy.zeros((2, 2, 3), dtype=numpy.uint8),
-                numpy.ones((2, 2, 3), dtype=numpy.uint8) * 255,
-                numpy.ones((2, 2, 3), dtype=numpy.uint8) * 255,
-            ]
-            # Start with no motion
-            for i in range(6):
-                yield Frame(data[0], time=1466084606. + i)
-            # Motion starts on 6th frame at time 1466084612.
-            for i in range(6, 20):
-                yield Frame(data[i % len(data)], time=1466084606. + i)
-
-        def draw(self, *_args, **_kwargs):
-            pass
+    FRAMES = []
+    data = [
+        numpy.zeros((2, 2, 3), dtype=numpy.uint8),
+        numpy.zeros((2, 2, 3), dtype=numpy.uint8),
+        numpy.ones((2, 2, 3), dtype=numpy.uint8) * 255,
+        numpy.ones((2, 2, 3), dtype=numpy.uint8) * 255,
+    ]
+    # Start with no motion
+    for i in range(6):
+        FRAMES.append(Frame(data[0], time=1466084606. + i))
+    # Motion starts on 6th frame at time 1466084612.
+    for i in range(6, 20):
+        FRAMES.append(Frame(data[i % len(data)], time=1466084606. + i))
 
-    dut = DeviceUnderTest(display=FakeDisplay(), sink_pipeline=NoSinkPipeline())
+    class FakeDisplay(object):
+        def get_frame(self, timeout_secs=10, since=0):  # pylint: disable=unused-argument
+            for f in FRAMES:
+                if f.time > since:
+                    return f
+            f = FRAMES[-1].copy()
+            f.time = since + 1
+            return f
+
+    class FakeTime(object):
+        def __init__(self, now):
+            self._now = now
+
+        def time(self):
+            return self._now
+
+        def sleep(self, duration):
+            self._now += duration
+
+    dut = DeviceUnderTest(display=FakeDisplay(), sink_pipeline=NoSinkPipeline(),
+                          _time=FakeTime(FRAMES[0].time))
     dut.get_frame = lambda: None
```

Comment:
Do we still need `dut.get_frame = lambda: None`?

Reply:
Probably not.  Removed.